src/matrixrtc/CallMembership.ts

```diff
@@ -472,40 +276,29 @@ export class CallMembership {
     public getTransport(oldestMembership: CallMembership): Transport | undefined {
         const { kind, data } = this.membershipData;
         switch (kind) {
-            case "rtc":
+            case MembershipKind.RTC:
                 return data.rtc_transports[0];
-            case "session":
-                switch (data.focus_active.focus_selection) {
-                    case "multi_sfu":
-                        return data.foci_preferred[0];
-                    case "oldest_membership":
-                        if (CallMembership.equal(this, oldestMembership)) return data.foci_preferred[0];
-                        if (oldestMembership !== undefined) return oldestMembership.getTransport(oldestMembership);
-                        break;
+            case MembershipKind.Session:
+                if (data.focus_active.focus_selection === "oldest_membership") {
+                    // For legacy events we only support "oldest_membership"
+                    if (CallMembership.equal(this, oldestMembership)) return data.foci_preferred[0];
+                    if (oldestMembership !== undefined) return oldestMembership.getTransport(oldestMembership);
                 }
+                break;
         }
         return undefined;
     }
```

Comment:
In general the changes to the call membership are much more straight forward than what I anticipated. this seems to be the main functional change.

Is there anything else in the code that gets more complicated because of this?

Reply:
Not really, I only removed things which didn't break other files or ECall / EWeb. I think the conflicts with the other PRs may be high.